interview-questions.md

```diff
@@ -4,64 +4,67 @@ Answer the following questions. First, without external resources. Challenge you
 
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: console.log() prints your code to the screen, tells the computer to run your code.  return is part of a function that specifies exactly what you want the function to return when you 
 
-  Researched answer:
+  Researched answer: console.log() will display the parameter passed to the log method in the console window.  A return statement makes the function stop executing and return the specified value.
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: An argument is what you are trying to find so when you write out a function, it's what would go in the parantheses as the placeholder.  
+  Ex. const test = (argument) => {
+  }
 
-  Researched answer:
+  Researched answer: Function arguments are the real values passed to (and received by) the function.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: Proper pair programming is between two people (there's another term for groups of 3 or more).  One person is the "Driver" the other is the "Navigator".  The "Driver" is the person who is at the keyboard and tackles the obstacles as they come.  "Navigator" looks at the road ahead and helps keep the team focused on what the objective is.  Benefits would be consistent code, "two heads are better than one," more focus.
 
-  Researched answer:
+  Researched answer: Pair programming consists of two programmers sharing a single workstation.  The programmer at the keyboard is the "driver", the other is the "navigator" who is actively involed in the programming as well but their focus is more of an overall approach.  Swap at appropriate increments.
 
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
-  Your answer:
+  Your answer: TDD - Test Driven Development - probably got that acronym wrong, but I know it's when we use a testing program to check our code.  Jest would be an example of a program that does TDD. 
 
-  Researched answer:
+  Researched answer: Test-driven development is a programming methodology with which one can tackle the design, implementation, and testing of units of code, and to some extent the expected funcitonality of a program.  Has three cycles: writing a failing test, making the test pass, and then refactoring the implemenation.
 
 
 
 7. What is something we did in class this week you found helpful?  
 
-  Your answer:
+  Your answer: Everything we did this week in class was helpful in some way or another.  I'm glad I'm finally learning what exactly git and GitHub do and are capable of.  I was able to set up git on a PC I use to pratice code using the terminal which was pretty awesome since I could understand what it was doing and the commands I used to install it made sense to me instead of just looking like a bunch of random words. 
```

Comment:
git is such a cool thing to be knowledgable about! makes you feel like a real hacker ;)